lib/phase/structural.js

```diff
@@ -15,20 +15,12 @@ function checkAdditionalRules(data) {
     const error = [];
 
     Object.keys(data.jobs).forEach((job) => {
-        const stepList = [];
         let firstUserTeardownStep = -1;
 
         steps = data.jobs[job].steps;
 
         if (steps) { // if there are user-defined steps
             for (let i = 0; i < steps.length; i += 1) {
-                const stepName = Object.keys(steps[i])[0];
-
-                if ((typeof (steps[i]) === 'object') && (stepList.includes(stepName))) {
-                    error.push(new Error(`Job ${job} has duplicate step: ${stepName}`));
```

Comment:
It was fixed to be detected by JOI.
https://github.com/screwdriver-cd/screwdriver/issues/1317